### ISSUE TITLE
"Type[arg[slice]]" is not iterable

### ISSUE THREAD

**Issue by htarnacki** [microsoft/pylance-release#59]
hi,

following code:
```
class arg:

    def __getitem__(self, idx: Union[int, slice]) -> Any:
        return sys.argv[idx]

    def __len__(self) -> int:
        return len(sys.argv)

    def __call__(self, idx: int) -> Any:
        if idx == -1:
            arg, = sys.argv[-1:] or [None]
        else:
            arg, = sys.argv[idx:idx + 1] or [None]
        return arg


# make it singletone
arg = arg()
```

produces errors in pylance:
![Zrzut_ekranu_2020-07-06-103935](https://user-images.githubusercontent.com/49711873/86573944-5ec0e480-bf75-11ea-9cef-161cb30fa2aa.png)

Pylance do not see that class is instantiated. Changing class name to "Arg" solves the problems.

**Comment by erictraut** [comment]
Thanks for reporting the issue.

Here's what's happening here. When you declare a class called `arg`, you're telling the type checker that the type of the symbol `arg` is a class. You're later trying to assign that same symbol a value that is not a class, so it's a type violation. The type checker would normally report this as an error, but by default Pylance has typeCheckingMode set to 'off', so that error is suppressed. Because it's suppressed, subsequent errors based on this type assumption don't make sense.

I think the correct fix here is to allow assignments to override the declared type of a symbol if typeCheckingMode is 'off'. In your case, it would allow the `arg` symbol to take on the type of an instance even though it is declared as having a class type above.

This change will be in an upcoming version of Pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.7.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202070-9-july-2020
